pkg/storaged/smart-details.jsx

```diff
@@ -0,0 +1,120 @@
+    };
+
+    const actions = [
+        <DropdownItem key="run-short-test"
+                      isDisabled={smartSelftestStatus === "inprogress"}
+                      onClick={() => { setKebabOpen(false); runSmartTest('short') }}>
+            {_("Run short test")}
+        </DropdownItem>,
+        <DropdownItem key="run-extended-test"
+                      isDisabled={smartSelftestStatus === "inprogress"}
+                      onClick={() => { setKebabOpen(false); runSmartTest('extended') }}>
+            {_("Run extended test")}
+        </DropdownItem>,
+        <DropdownItem key="run-conveyance-test"
+                      isDisabled={smartSelftestStatus === "inprogress"}
+                      onClick={() => { setKebabOpen(false); runSmartTest('conveyance') }}>
+            {_("Run conveyance test")}
+        </DropdownItem>,
+    ];
+
+    if (smartInfo.SmartSelftestStatus === "inprogress") {
+        actions.push(
+            <DropdownItem key="abort-smart-test"
+                          onClick={() => { setKebabOpen(false); abortSmartTest('conveyance') }}>
+                {_("Abort test")}
+            </DropdownItem>,
+        );
+    }
+
+    return (
+        <Dropdown toggle={<KebabToggle onToggle={(_, isOpen) => setKebabOpen(isOpen)} />}
+                isPlain
+                isOpen={isKebabOpen}
+                position="right"
+                id="smart-actions"
+                dropdownItems={actions} />
+    );
+};
+
+export const SmartDetails = ({ smartInfo }) => {
+    const SmartDetailRow = ({ title, value }) => {
+        if (value === undefined)
+            return null;
+
+        return (
+            <DescriptionListGroup>
+                <DescriptionListTerm>{title}</DescriptionListTerm>
+                <DescriptionListDescription>{value}</DescriptionListDescription>
+            </DescriptionListGroup>
+        );
+    };
+
+    return (
+        <Card>
+            <CardHeader actions={{ actions: <SmartActions smartInfo={smartInfo} /> }}>
+                <CardTitle component="h2">{_("S.M.A.R.T")}</CardTitle>
+            </CardHeader>
+            <CardBody>
+                <DescriptionList isHorizontal horizontalTermWidthModifier={{ default: '20ch' }}>
+                    <SmartDetailRow title={_("Power on hours")} value={cockpit.format(_("$0 hours"), Math.round(smartInfo.SmartPowerOnSeconds / 3600))} />
+                    <SmartDetailRow title={_("Last updated")} value={timeformat.dateTime(new Date(smartInfo.SmartUpdated * 1000))} />
+                    <SmartDetailRow title={_("Smart selftest status")} value={selftestStatusDescription[smartInfo.SmartSelftestStatus]} />
+                    <SmartDetailRow title={_("Number of bad sectors")} value={smartInfo.SmartNumBadSectors} />
+                    <SmartDetailRow title={_("Atributes failing")} value={smartInfo.SmartNumAttributesFailing} />
+                </DescriptionList>
+            </CardBody>
+        </Card>
```

Comment:
These 15 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19103-20230717-115949-00eb709f-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)